### PR TITLE
Update root-cert-stats to use runtime.sendMessage #542

### LINKED ISSUE #542: Update root-cert-stats to use runtime.sendMessage
### What information was incorrect, unhelpful, or incomplete?

This example use getBackgroundPage(). It's bad practice to use getBackgroundPage(), e.g. in a private browsing window that logic breaks. If you can fix the example to use what I just described, it would be great.

### What did you expect to see?

This example should use runtime.sendMessage from popup.js + runtime.onMessage in the background page to send the data back. 

### Do you have any supporting links, references, or citations?

See https://github.com/mdn/webextensions-examples/issues/536#issuecomment-1741871260

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by ibzimh**
PR for Issue: Update root-cert-stats to use runtime.sendMessage #542 

### Description

I've added functionality for `runtime.sendMessage` and `runtime.onMessage` and it works on my browser. I had to move the table displaying into a different function; i thought it would be easier to read than to put it in the response callback function. 

If there's something which is incorrect or should be changed, let me know! 

### Related issues and pull requests

"Fixes #542"

### CODE REVIEW

**Comment by rebloor** [comment]
@rpl I have exercised and confirmed the code appears to work as expected.
image
Can you review the coding?

**Review by Rob--W — APPROVED** [review]
Looks good, thanks!

I recently merged a linter workflow integration, I'll try to test that with your PR.

File: root-cert-stats/popup.js [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
-if (entries.length > 0) {
+browser.runtime.sendMessage({ action: "getRootCertStats" }, response => {
+  displayTable(response.rootCertStats);
+});
 
-  let noData = document.querySelector(".no-data");
-  noData.classList.add("hidden");
-  let entryTable = document.querySelector(".root-cert-table");
-  entryTable.classList.remove("hidden");
+const backgroundPage = browser.extension.getBackgroundPage();
```
Comment (Rob--W) [inline root-cert-stats/popup.js:10]:
```suggestion
```

`backgroundPage` is no longer used, let's remove it.

**Comment by rebloor** [comment]
@ibzimh many thanks for your help on this one

**Pull request merged**